### PR TITLE
feat: Add resource type to user metadata of a blob #465

### PR DESCRIPTION
**Pull request by PMitrafanau**
Issue: #465

### CODE REVIEW

File: src/main/java/com/epam/aidial/core/service/ResourceService.java [hunk 1]
```diff
@@ -762,6 +774,9 @@ private static Map<String, String> toUserMetadata(String etag, Long createdAt, L
         if (updatedAt != null) {
             metadata.put(ResourceUtil.UPDATED_AT_ATTRIBUTE, Long.toString(updatedAt));
         }
+        if (resourceType != null) {
```
Comment (akurnosau) [inline src/main/java/com/epam/aidial/core/service/ResourceService.java:777]: Question: is it possible that Resource will not have ResourceType? I thought it is used everywhere at least to build a path.

Comment (akurnosau) [reply src/main/java/com/epam/aidial/core/service/ResourceService.java:777]: Discussed offline. That check makes sense.

**Review by akurnosau — APPROVED** [review]

File: src/main/java/com/epam/aidial/core/service/ResourceService.java (hunk 1 above)
Comment (PMitrafanau) [reply src/main/java/com/epam/aidial/core/service/ResourceService.java:777]: For the context: I've traced all the scenarios I could and indeed the `resourceType` seems to be non-nullable here. And so is `updatedAt`. But as there is still a check for the `updatedAt` it means I could miss something or it's there just-in-case. With this in mind, I've decided to put the same check for the `resourceType`.

**Comment by sdryapko** [comment]
@astsiapanay  please review

**Review by astsiapanay — APPROVED** [review]

**Pull request merged**